odbc/src/api/data.rs

```diff
@@ -115,6 +115,22 @@ impl IntoCData for Bson {
             Bson::Int64(i) => Ok(i.to_le_bytes().to_vec()),
             Bson::Binary(b) => Ok(b.bytes),
             Bson::Decimal128(d) => Ok(d.bytes().to_vec()),
+            Bson::Document(d) => Ok(d.to_string().into_bytes()),
+            Bson::Array(a) => Ok(serde_json::to_string(
+                &a.into_iter().map(|b| b.to_json()).collect::<Vec<String>>(),
+            )
+            .unwrap()
+            .into_bytes()),
+            Bson::ObjectId(o) => Ok(o.to_string().into_bytes()),
+            Bson::Undefined => Ok(char::from(0).to_string().into_bytes()),
+            Bson::Null => Ok(char::from(0).to_string().into_bytes()),
```

Comment:
I'm not entirely sure about representing these as `0`.  Since we return MinKey and MaxKey as their name, it might make sense to do the same.  
@nbagnard what would be a good way to represent/display the Null and Undefined values here?

Reply:
This is taken care of one step before in format_bson_data : Bson::Null and Bson::Undefined will show NULL (as in data is Null). For Bson::Null that because SQL does not make the distinction between Null and missing. For undefined, it's as good a representation as any ;)